### PR TITLE
Move nevergrad to orion repo

### PR DESCRIPTION
**Pull request by Delaunay**
_Hi there! Thank you for contributing. Feel free to use this pull request template; It helps us reviewing your work at its true value._

_Please remove the instructions in italics before posting the pull request :)._

# Description
_Describe what is the purpose of this pull request and why it should be integrated to the repository. 
When your changes modify the current behavior, explain why your solution is better._

_If it solves a GitHub issue, be sure to [link it](https://help.github.com/en/github/writing-on-github/autolinked-references-and-urls#issues-and-pull-requests)._

# Changes
_Give an overview of the suggested solution._

# Checklist
_This is simply a reminder of what we are going to look for before merging your code._

_Add an `x` in the boxes that apply._
_If you're unsure about any of them, don't hesitate to ask. We're here to help!_ 
_You can also fill these out after creating the PR if it's a work in progress (be sure to publish the PR as a draft then)_

## Tests
- [ ] I added corresponding tests for bug fixes and new features. If possible, the tests fail without the changes
- [ ] All new and existing tests are passing (`$ tox -e py38`; replace `38` by your Python version if necessary)

## Documentation
- [ ] I have updated the relevant documentation related to my changes

## Quality
- [ ] I have read the [CONTRIBUTING](https://github.com/Epistimio/orion/blob/develop/CONTRIBUTING.md) doc
- [ ] My commits messages follow [this format](https://chris.beams.io/posts/git-commit/)
- [ ] My code follows the style guidelines (`$ tox -e lint`)

# Further comments
_Please include any additional information or comment that you feel will be helpful to the review of this pull request._

### CODE REVIEW

File: tests/unittests/algo/nevergrad/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+## Python requirements for running tests.
+##
+##    pip install --requirement=requirements.txt
+##
+pytest>=3.5
```
Comment (bouthilx) [inline tests/unittests/algo/nevergrad/requirements.txt:5]: Ideally these requirements would be in the algo extra dependencies in setup.py or in test extra dependencies in setup.py.

File: src/orion/testing/algo.py [hunk 2]
```diff
@@ -355,9 +355,11 @@ def assert_dim_type_supported(self, mocker, num, attr, test_space):
         space = self.create_space(self.update_space(test_space))
         algo = self.create_algo(space=space)
 
+        print(space)
```
Comment (bouthilx) [inline src/orion/testing/algo.py:358]:
```suggestion
```

File: src/orion/testing/algo.py [hunk 3]
```diff
@@ -355,9 +355,11 @@ def assert_dim_type_supported(self, mocker, num, attr, test_space):
...
         spy = self.spy_phase(mocker, num, algo, attr)
-
         trials = algo.suggest(1)
+        print(trials)
```
Comment (bouthilx) [inline src/orion/testing/algo.py:361]:
```suggestion
```

**Review by bouthilx — CHANGES_REQUESTED** [review]
LGTM. Only misses documentation. You can take PR #866 as a reference.

File: src/orion/algo/nevergradoptimizer.py [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+class NevergradOptimizer(BaseAlgorithm):
+    """Wraps the nevergrad library to expose its algorithm to orion
+
+    Parameters
+    ----------
+    space: `orion.algo.space.Space`
+        Optimisation space with priors for each dimension.
+    seed: None, int or sequence of int
+        Seed for the random number generator used to sample new trials.
+        Default: ``None``
```
Comment (bouthilx) [inline src/orion/algo/nevergradoptimizer.py:164]: Missing model_name, budget and num_workers.

File: src/orion/algo/nevergradoptimizer.py [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+            )
+            return False
+
+        return True
+
+    def suggest(self, num):
+        """Suggest a number of new sets of parameters.
+
+        TODO: document how suggest work for this algo
+
```
Comment (bouthilx) [inline src/orion/algo/nevergradoptimizer.py:299]:
```suggestion
```

File: src/orion/algo/nevergradoptimizer.py [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+        -------
+        list of trials or None
+            A list of trials representing values suggested by the algorithm. The algorithm may opt
+            out if it cannot make a good suggestion at the moment (it may be waiting for other
+            trials to complete), in which case it will return None.
+
+
+        Notes
+        -----
+        New parameters must be compliant with the problem's domain `orion.algo.space.Space`.
```
Comment (bouthilx) [inline src/orion/algo/nevergradoptimizer.py:316]:
```suggestion
```

File: src/orion/algo/nevergradoptimizer.py (hunk 4 above)
Comment (Delaunay) [reply src/orion/algo/nevergradoptimizer.py:170]: added

**Review by bouthilx — APPROVED** [review]
Thanks!

**Pull request merged**